src/main/java/org/jenkinsci/test/acceptance/po/JobsMixIn.java

```diff
@@ -58,7 +58,14 @@ public Object call() throws Exception {
     }
 
     public <T extends TopLevelItem> T get(Class<T> type, String name) {
-        return newInstance(type, injector, url("job/%s/", name), name);
+        try {
+            assert getContext() != null;
+            return newInstance(type, getContext(), url("job/%s/", name), name);
+        } catch (AssertionError e) {
```

Comment:
I do not like this as we throw the exception all over the place. We should either fix all the implementations and accept external client will be broken or make the `newInstance` to throw more specific exception to implement this fallback.